File: opendaylight/md-sal/sal-zeromq-connector/src/main/java/org/opendaylight/controller/sal/connector/zeromq/Activator.java
package org.opendaylight.controller.sal.connector.zeromq;

import org.opendaylight.controller.sal.core.api.AbstractProvider;
import org.opendaylight.controller.sal.core.api.Broker.ProviderSession;
import org.osgi.framework.BundleContext;

public class Activator extends AbstractProvider {
    
    ZeroMqRpcRouter router;
    
    @Override
    public void onSessionInitiated(ProviderSession session) {
        router = ZeroMqRpcRouter.getInstance();
        router.setBrokerSession(session);
        router.start();
    }
    
    @Override
    protected void stopImpl(BundleContext context) {
       router.stop();
    }

}

File: runtime/clustering/src/main/resources/org/opendaylight/controller/rest/Activator.scala
package org.opendaylight.controller.rest

import akka.actor.{ Props, ActorSystem }
import org.osgi.framework.BundleContext
import akka.osgi.ActorSystemActivator

class Activator extends ActirSystenActivator {

    def configure(context: BundleContext, system: ActorSystem) {
        // optionally register the ActorSystem in the OSGi Service Registry
        registerService(context, system)

        val rest = system.actorOf(Props[Rest], name = "rest")
        someActor ! SomeMessage
    }

    def start(context: BundleContext) {
      println("REST Service started")
      val system = ActorSystem("features-api")
/*
      // Initialize REST Manager
      val rest = system.actorOf(Props[Rest], name = "rest")

      }
*/
    def stop(context: BundleContext) {}
}